tests/test_library/include/simpleton.hpp

```diff
@@ -1,7 +1,20 @@
 #ifndef H5_TEST_SIMPLETON_HPP
 #define H5_TEST_SIMPLETON_HPP
 
+#include <highfive/H5Attribute.hpp>
```

Comment:
We already have tests that #include headers one by one, I don't see as necessary all these imports here. Plus, most headers will import recursively a lot of others. 
Maybe in the future we create a HighFive.hpp master header.

Reply:
I'd rather do this explicitly and dumb… trying to be as fool-proof as possible.

Reply:
I meant the tests using `tests_import_public_headers.cpp` which cover all the cases here, but it's not a blocker either way.

Reply:
I'd be up for a master header, the point here is to catch any missing `inline`s that show up when all headers are included in two different compilation units. Thus it may be a little different from the public import test…